src/main/java/com/felixgrund/codeshovel/visitors/TypeScriptVisitor.java

```diff
@@ -0,0 +1,2289 @@
+package com.felixgrund.codeshovel.visitors;
+
+import com.eclipsesource.v8.*;
+import com.eclipsesource.v8.utils.MemoryManager;
+import com.felixgrund.codeshovel.util.TypeScript;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TypeScriptVisitor {
+
+	protected final V8Object ts;
+	private V8Object syntaxKind;
+	private MemoryManager scope;
+
+	private static final Map<String, Integer> syntaxKindCache = new HashMap<String, Integer>();
+
+	public TypeScriptVisitor() {
+		ts = TypeScript.getInstance().getTS();
+		init();
+	}
+
+	public void visit(String source) {
+		visit(getSource(source));
+		clear();
+	}
+
+	protected V8Object getSource(String source) {
+		V8Object scriptTargetEnum = ts.getObject("ScriptTarget");
+		V8Object scriptKindEnum = ts.getObject("ScriptKind");
+
+		String name = "temp.ts";
+		// TODO is this the script target we want to be using?
```

Comment:
I'm pretty sure the Script Target will have no effect on this visitor, but I would love it if someone could confirm